framework/src/main/java/org/checkerframework/framework/stub/StubTypes.java

```diff
@@ -202,15 +202,31 @@ public void parseStubFiles() {
                                         + ".class, but is at the top level of a jar file: "
                                         + topLevelResource);
                     } else {
-                        checker.message(
-                                Kind.WARNING,
-                                "Did not find stub file "
-                                        + stubPath
-                                        + " on classpath or within directory "
-                                        + new File(stubPath).getAbsolutePath()
-                                        + (stubPathFull.equals(stubPath)
-                                                ? ""
-                                                : (" or at " + stubPathFull)));
+                        // When using a compound checker, the target stub file may be found by the
+                        // current checker's parent checkers. Also check this to avoid a false
+                        // warning.
+                        SourceChecker parentChecker = checker.parentChecker;
+                        boolean findByParentCheckers = false;
+                        while (parentChecker != null) {
```

Comment:
I think you should try to pull this check out before the first use of `checker.getClass()` on line 192 and simply perform the same logic for either the first checker or one of the parent checkers.
This would include checking for the `topLevelResource` for each possible checker.